src/classes/PS_IntegrationService.cls

```diff
@@ -35,16 +35,108 @@
+    public static final String DELIMETER_CONST = '::';
+
+    private static final Map<String, String> fieldMappingConstants = new Map<String, String>{
+            KEY_CONST => 'Key__c',
+            VALUE_CONST => 'Value__c',
+            IS_SECRET_CONST => 'Is_Secret__c',
+            SERVICE_CONST => 'Service__c'
+    };
+
+
     /*******************************************************************************************************
     * @description Provides an endpoint for the Payment Services Integrations package
     * to call when a Customer registers, and pass parameters containing the secrets
     * needed for calls to Payment Services endpoints(Tokenize, Purchase) and creating JWTs
-    * @param secrets contains secrets, such as the api key for creating a JWT
+    * @param configParams contains secrets, such as the api key for creating a JWT
     */
     global static void setConfiguration(Map<String, List<Map<String, String>>> configParams) {
-        // TODO This method needs to be implemented after we decide on the storage mechanism
-        // for the onboarding data
 
+        if (configParams == null || configParams.isEmpty()) {
+            throw new IntegrationServiceException(INVALID_CONFIG_PARAMS);
+        }
+
+        List<Payment_Services_Configuration__c> configurations = new List<Payment_Services_Configuration__c>();
+
+        for (String service : configParams.keySet()) {
+            for (Map<String, String> valueMap : configParams.get(service)) {
+                configurations.add(createConfiguration(service, valueMap));
+            }
+        }
+
+        try {
+            upsert configurations Service_Key__c;
+        } catch (Exception ex) {
+            throw new IntegrationServiceException(ELEVATE_ONBOARDING_ERROR + ex.getMessage());
+        }
+
+    }
+
+    /******************************************************************************************************
+    * @description This method builds the a configuration object based on key value pair configuration
+    * parameters and the configuration service type
+    * @param service Configuration service type, ex: payments, makana
+    * @param keyValuePairs
+    *
+    * @return Payment Services Configuration object
+    */
+    public static Payment_Services_Configuration__c createConfiguration(
+            String service, Map<String, String> keyValuePairs) {
+
+        Payment_Services_Configuration__c configuration = new Payment_Services_Configuration__c();
+        configuration = setValue(configuration, SERVICE_CONST, service);
+
+        for (String key : keyValuePairs.keySet()) {
+            Object value = keyValuePairs.get(key);
+            configuration = setValue(configuration, key, value);
+        }
+
+        configuration.Service_Key__c = configuration.Service__c + DELIMETER_CONST + configuration.Key__c;
+        return configuration;
     }
+
+
+    public static Payment_Services_Configuration__c setValue(
```

Comment:
Should this be private instead of public? Do we expect to call it outside of this class?